pkgs/tools/system/osquery/default.nix

```
@@ -0,0 +1,82 @@
+{ pkgs }:
+
+with pkgs;
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
+  opensslArchive =
+    let
+      # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4.
+      version = "1.1.1q";
+      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
+    in
+    fetchurl {
+      inherit sha256;
+      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+    };
```

We have an openssl_1_1. Please use that.


```suggestion
  opensslArchive =
    let
      # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4.
      version = "1.1.1q";
    in
    fetchurl {
      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
    };
```

---

The osquery makefile seems to need the source of openssl and not the library itself. Do you mean we should get the source from `pkgs.openssl_1_1.src` instead of downloading an openssl archive?

---

That would be preferable, to avoid putting a sort of double maintenance burden on that OpenSSL setup.

Note there is also `pkgs.srcOnly` which applies potentially present patches to the source as well. I don't know if we carry any OpenSSL patches, but might be worth checking.

---

Also because of the usage of openssl 1.1 we might need to mark this as insecure soon.